app/utils/formInitialization.ts

```diff
@@ -1,40 +1,34 @@
-import type { FormValues, ValidationFunction } from '../types/signupForm';
-
-interface ConfigItem {
-  field: keyof FormValues;
-  validate?: ValidationFunction;
-}
+import type {
+  ConfigItem,
+  FormValues,
+  ValidationFunction,
+} from '../types/signupForm';
 
 /**
  * Generates initial form values based on the provided config.
- * @param config - Array of ConfigItem objects.
- * @returns Object with field names as keys and empty strings as values.
+ * @param {ConfigItem[]} config - Array of ConfigItem objects.
+ * @returns {FormValues} Object with field names as keys and empty strings as values.
  */
 export const generateInitialValues = (config: ConfigItem[]): FormValues => {
-  const initialValues = {} as FormValues;
-
-  config.forEach((item) => {
+  return config.reduce((initialValues, item) => {
     initialValues[item.field] = '';
-  });
-
-  return initialValues;
+    return initialValues;
+  }, {} as FormValues);
 };
 
 /**
  * Generates validation rules based on the provided config.
- * @param config - Array of ConfigItem objects.
- * @returns Object with field names as keys and validation functions as values.
+ * @param {ConfigItem[]} config - Array of ConfigItem objects.
+ * @returns {Record<keyof FormValues, ValidationFunction | undefined>} Object with field names as mantine keys and validation functions as values.
  */
 export const generateValidationRules = (
   config: ConfigItem[]
-): Record<string, ValidationFunction> => {
-  const validationRules: Record<string, ValidationFunction> = {};
-
-  config.forEach((item) => {
-    if (item.validate) {
-      validationRules[item.field] = item.validate;
-    }
-  });
-
-  return validationRules;
+): Record<keyof FormValues, ValidationFunction | undefined> => {
+  return config.reduce(
+    (validationRules, item) => {
+      validationRules[item.field] = item.validate ?? undefined;
+      return validationRules;
+    },
+    {} as Record<keyof FormValues, ValidationFunction | undefined>
```

Comment:
Thanks for your suggestions about the two helper functions! Honestly, I'm not entirely satisfied with them as they seem to be performing similar and relatively straightforward tasks.

I fully agree with you on the importance of type safety. However, I believe that specifying all the default fields goes against the purpose of using a dynamic value generator function - which is totally ok but if we are going that route we may use a simple subject instead of using a function. 

I'm also considering if it might be better to create a more generic function for initializing the field props.🤔